apps/dashboard/src/hooks/use-promote-workflow.ts

```diff
@@ -0,0 +1,114 @@
+import { getV2, NovuApiError } from '@/api/api.client';
+import { promoteWorkflow } from '@/api/workflows';
+import { promoteToast } from '@/components/primitives/sonner-helpers';
+import { useEnvironment } from '@/context/environment/hooks';
+import { WorkflowListResponseDto, WorkflowOriginEnum, WorkflowResponseDto, WorkflowStatusEnum } from '@novu/shared';
+import { useMutation } from '@tanstack/react-query';
+import { useCallback, useMemo, useState } from 'react';
+import { toast } from 'sonner';
+
+const PRODUCTION_ENVIRONMENT = 'Production' as const;
+
+export function usePromoteWorkflow(workflow: WorkflowListResponseDto) {
+  const { environments, switchEnvironment, currentEnvironment } = useEnvironment();
+  const [isLoading, setIsLoading] = useState(false);
+  let loadingToast: string | number | undefined = undefined;
+
+  const isPromotable = useMemo(
+    () =>
+      workflow.origin === WorkflowOriginEnum.NOVU_CLOUD &&
+      workflow.status !== WorkflowStatusEnum.ERROR &&
+      currentEnvironment?.name !== PRODUCTION_ENVIRONMENT,
+    [workflow.origin, workflow.status, currentEnvironment?.name]
+  );
+
+  const getProductionEnvironmentId = useCallback(() => {
+    return environments?.find((environment) => environment.name === PRODUCTION_ENVIRONMENT)?._id || '';
+  }, [environments]);
+
+  const getTooltipContent = () => {
+    if (workflow.origin === WorkflowOriginEnum.EXTERNAL) {
+      return 'External workflows cannot be promoted to Production using dashboard.';
+    }
+
+    if (workflow.origin === WorkflowOriginEnum.NOVU_CLOUD_V1) {
+      return 'V1 workflows cannot be promoted to Production using dashboard. Please visit the legacy portal.';
+    }
+
+    if (workflow.status === WorkflowStatusEnum.ERROR) {
+      return 'This workflow has errors and cannot be promoted to Production. Please fix the errors first.';
+    }
+  };
+
+  const safePromote = async () => {
+    try {
+      if (await isWorkflowInProduction()) {
+        return { needsConfirmation: true };
+      }
+    } catch (error) {
+      if (error instanceof NovuApiError && error.status === 404) {
+        return { needsConfirmation: false };
+      }
+
+      toast.error('Failed to promote workflow', {
+        description: error instanceof Error ? error.message : 'There was an error promoting the workflow.',
+      });
+
+      return { needsConfirmation: false };
+    }
+
+    await promoteWorkflowMutation();
+
+    return { needsConfirmation: false };
+  };
```

Comment:
I would suggest you to split this hook to two: `useIsWorkflowInProduction` and `usePromoteWorkflow`.

In the first one you can use `useMutation` and pass `onSuccess` and `onError` callbacks where you can handle all the logic to show/not the confirmation modal, and call the `promoteWorkflowMutation`.

Reply:
I think splitting this to 2 hooks would cause more confusion, since these 2 hooks will always be used together. 

If `useIsWorkflowInProduction` would only check if workflow is in production then I thinks its too small usecase for a hook, since its just 1 API call and as of now it won't be reused. If it would also include the display of confirmation modal, then it needs to be always used directly with `usePromoteWorkflow` only - its a part of promotion logic not a standalone hook.

I think the simplest option is what we have now - just one not that complicated (< 200 lines) hook, which encapsulates everything the user needs to promote workflow in any place/context

Reply:
I'm fine with that, but please just use the `useMutation` for the `isWorkflowInProduction` check, as we don't use the API directly anywhere but always with the `react-query` hooks ;)